lib/phlex/context.rb

```diff
@@ -23,6 +24,15 @@ def target_fragments(fragments)
 		@fragments = fragments.to_h { |it| [it, true] }
 	end
 
+	def around_render
+		return yield if @halt_signal
+
+		catch do |signal|
+			@halt_signal = signal
+			yield
+		end
```

Comment:
I should have mentioned that I tried this and found it slowed the benchmark significantly.